lib/msal-browser/src/interaction_client/StandardInteractionClient.ts

```diff
@@ -69,6 +69,29 @@ export abstract class StandardInteractionClient extends BaseInteractionClient {
             ...logoutRequest
         };
 
+        /**
+         * Set logout_hint to be login_hint from ID Token Claims if present
+         * and logoutHint attribute wasn't manually set in logout request
+         */
+        if (logoutRequest) {
+            // If logoutHint isn't set and an account was passed in, try to extract logoutHint from ID Token Claims
+            if (!logoutRequest.logoutHint) {
+                if(logoutRequest.account) {
```

Comment:
Should we use the `activeAccount` if no account is provided on the request?

Reply:
It depends, passing an account in is more "intentional". If we default to the active account, then apps can't really "opt into" having the prompt.

Reply:
I thought about this as well, but given that currently, if you dont pass an account but do have activeAccount set, MSAL clears the cache for all accounts, not just the active account, I figured we should treat this the team and not only logout the active account from the server.

Reply:
Flat if statements

```js
getLogoutHint(logoutRequest) {
    if (!logoutRequest) {
      return {
        message: "logoutHint will not be set since no logout request was configured",
        logouthint: null
      }
    }

    if (logoutRequest.logoutHint) {
      return {
        message: "logoutHint has already been set in logoutRequest",
        logoutHint: null
      }
    }

    if (!logoutRequest.account) {
      return {
        message: "logoutHint was not set and account was not passed into logout request, logoutHint will not be set",
        logoutHint: null
      }
    }

    const logoutHint = this.getLogoutHintFromIdTokenClaims(
      logoutRequest.account
    );

    if (!logoutHint) {
      return {
        message: null, // are you missing a message here on purpose?
        logoutHint: null
      }
    }

    return {
      message: "Setting logoutHint to login_hint ID Token Claim value for the account provided",
      logoutHint
    }
  },
  
  ...
  const { message, logoutHint } = this.getLogoutHint(logoutRequest);

  if (message) this.logger.verbose(message);
  if (logoutHint) validLogoutRequest.logoutHint = logoutHint;
  ...